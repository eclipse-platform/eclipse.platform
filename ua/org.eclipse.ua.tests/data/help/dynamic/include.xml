<?xml version="1.0"?>
<!--
     Copyright (c) 2005, 2006 IBM Corporation and others.

     This program and the accompanying materials
     are made available under the terms of the Eclipse Public License 2.0
     which accompanies this distribution, and is available at
     https://www.eclipse.org/legal/epl-2.0/

     SPDX-License-Identifier: EPL-2.0
    
     Contributors:
         IBM Corporation - initial API and implementation
 -->

<root>

	<description>
		This file tests the include capability. Below are some includes that
		pull in content from other files.
	</description>
	
	<element id="element.1">
		This element is used to test includes within the same file.
	</element>
	
	<!--
	     Valid cases
	-->
	
	<!-- valid include -->
	<include path="/org.eclipse.ua.tests/data/help/dynamic/include/doc1.xml#paragraph.1"/>

	<!-- include from same document -->
	<include path="/org.eclipse.ua.tests/data/help/dynamic/include.xml#element.1"/>

	<!-- included element has yet more dynamic content -->
	<include path="/org.eclipse.ua.tests/data/help/dynamic/shared/doc1.xml#doc.1"/>

	<!-- legacy format: no slash at beginning, slash in place of # -->
	<include path="org.eclipse.ua.tests/data/help/dynamic/include/doc1.xml/paragraph.1"/>

	<!-- extra unknown parameters -->
	<include unknownparam="unknownvalue" path="/org.eclipse.ua.tests/data/help/dynamic/include/doc1.xml#paragraph.1" unknownparam2="unknownvalue2"/>

	<!-- extra unknown children -->
	<include path="/org.eclipse.ua.tests/data/help/dynamic/include/doc1.xml#paragraph.1">
	   <unknownElement>This element should be ignored</unknownElement>
	</include>

	<!--
	     Invalid cases
	-->

	<!-- missing path -->
	<include/>

	<!-- missing path; unknown parameter -->
	<include unknownparam="unknownvalue"/>
	
	<!-- invalid bundle id -->
	<include path="/org.eclipse.ua.tests.invalid/path/file.xml#id"/>

	<!-- invalid path -->
	<include path="/org.eclipse.ua.tests/invalid/doc1.xml#paragraph.1"/>

	<!-- invalid file -->
	<include path="/org.eclipse.ua.tests/data/help/dynamic/include/invalid.xml#paragraph.1"/>

	<!-- invalid id -->
	<include path="/org.eclipse.ua.tests/data/help/dynamic/include/doc1.xml#invalid"/>
	
</root>