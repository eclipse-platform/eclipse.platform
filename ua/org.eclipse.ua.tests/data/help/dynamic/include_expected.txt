<root>

	<description>
		This file tests the include capability. Below are some includes that
		pull in content from other files.
	</description>
	
	<element id="element.1">
		This element is used to test includes within the same file.
	</element>
	
	
	
	
	<paragraph id="paragraph.1">
		This is the first paragraph from
		/org.eclipse.ua.tests/data/help/dynamic/include/doc1.xml
	</paragraph>

	
	<element id="element.1">
		This element is used to test includes within the same file.
	</element>

	
	<doc id="doc.1">
	<description>
		This document contains a mixture of dynamic content and is used by other
		tests that incorporate outside content, to make sure it is processed.
	</description>
	<element id="element.1">
		This element has no replaces.
	</element>
	<element id="element.1">
		This is element 1 of doc2.xml
	</element>
	
	
	
	<element id="element.1">
		This is element 1 of doc2.xml
	</element>
	<element id="element.2">
		This is element 2 of doc2.xml
	</element><element id="element.3">
		This is element 3 of doc2.xml
	</element>
</doc>

	
	<paragraph id="paragraph.1">
		This is the first paragraph from
		/org.eclipse.ua.tests/data/help/dynamic/include/doc1.xml
	</paragraph>

	
	<paragraph id="paragraph.1">
		This is the first paragraph from
		/org.eclipse.ua.tests/data/help/dynamic/include/doc1.xml
	</paragraph>

	
	<paragraph id="paragraph.1">
		This is the first paragraph from
		/org.eclipse.ua.tests/data/help/dynamic/include/doc1.xml
	</paragraph>

	

	
	

	
	
	
	
	

	
	

	
	

	
	
	
</root>